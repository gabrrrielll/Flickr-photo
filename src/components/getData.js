import axios from "axios";

export const getdata = async (url) => {
    var response = await axios.get(url, {
        headers: {
            Accept: "application/json",
            "Content-Type": "application/json",
        },
        mode: "no-cors",
        credentials: "include",
    });

    return response;
};
