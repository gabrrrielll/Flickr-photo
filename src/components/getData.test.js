import axiosMock from "axios";
import { getdata } from "./getData";

jest.mock("axios");

test("loads data from API", async () => {
    const photos = { page: 1, pages: 6701, perpage: 100, total: "670055", photo: Array(100) };
    const res = { data: photos };
    let url = "https://api.flickr.com/services/rest/?method=flickr.photos.search&format=json&nojsoncallback=1&api_key=" + process.env.REACT_APP_SECRET_KEY + "&is_getty=1&page=1";

    const response = getdata(url);

    axiosMock.get.mockResolvedValueOnce(res);
    axiosMock.get.mockImplementation(() => Promise.resolve(res));

    expect(axiosMock.get).toHaveBeenCalledTimes(1);
    expect(response).toBeTruthy();
    expect(response).not.toBeNull();
    expect(response).not.toBe("error");
    expect(axiosMock.get).toMatchSnapshot();
    expect(response).toMatchSnapshot();
    getdata().then((data) => expect(data).toEqual(res));
});
