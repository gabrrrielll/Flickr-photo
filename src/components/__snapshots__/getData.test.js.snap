// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads data from API 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "https://api.flickr.com/services/rest/?method=flickr.photos.search&format=json&nojsoncallback=1&api_key=undefined&is_getty=1&page=1",
      Object {
        "credentials": "include",
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "mode": "no-cors",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`loads data from API 2`] = `Promise {}`;
